### PR TITLE
Login

### PR DESCRIPTION
**Pull request by robin-lillqvist**
Thomas helped us

### CODE REVIEW

**Comment by robin-lillqvist** [comment]
Updated Login branch to include sad path testing and cleaned up gemfile and rails helper

File: app/models/user.rb [hunk 1]
```diff
@@ -4,9 +4,12 @@ class User < ApplicationRecord
...
 
-  validates :name, length: { maximum: 10 }
-  validates :name, presence: true
-  validates_uniqueness_of :name
+
+         validates :name, length: { maximum: 10 }
+         validates :name, presence: true
+         validates :email, presence: true
+         validates :encrypted_password, presence: true
+        #  validates_uniqueness_of :name
```
Comment (faraznaeem) [inline app/models/user.rb]: remove comment

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -3,12 +3,12 @@
 <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
   <div class="field">
     <%= f.label :email %><br/>
-    <%= f.email_field :email, autofocus: true %>
+    <%= f.email_field :email, placeholder: "email@email.com", autofocus: true %>
```
Comment (faraznaeem) [inline app/views/devise/sessions/new.html.erb]: Please add better place holder

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -3,12 +3,12 @@
...
   </div>
 
   <div class="field">
     <%= f.label :password %><br/>
-    <%= f.password_field :password, autocomplete: "off" %>
+    <%= f.password_field :password, placeholder: "password", autocomplete: "off" %>
```
Comment (faraznaeem) [inline app/views/devise/sessions/new.html.erb]: please add better placeholder

File: spec/features/user_can_click_the_login_button_spec.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        end
+  
+        it 'expect to give message invalid password' do
+            fill_in 'Email', with: registered_user.email
+            fill_in 'Password', with: 'wrongpassword'
+            click_on 'Log in'
+            expect(page).to have_content 'Invalid Email or password'
+        end  
+    end    
+
```
Comment (faraznaeem) [inline spec/features/user_can_click_the_login_button_spec.rb]: remove extra lines

File: spec/features/user_can_click_the_login_button_spec.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
```
Comment (faraznaeem) [inline spec/features/user_can_click_the_login_button_spec.rb]: The name of the file should be in the lines of `user can login`

File: spec/features/user_can_see_a_inbox_spec.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
```
Comment (faraznaeem) [inline spec/features/user_can_see_a_inbox_spec.rb]: This feature is out of scope, but okay for now. In the future
1 feature = 1 PR

File: spec/features/user_can_see_login_form_spec.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
```
Comment (faraznaeem) [inline spec/features/user_can_see_login_form_spec.rb]: This feature file only checks for items, but no functionality, and is in my opinion not necessary as we check the functionality and the existence with the `login` feature

File: spec/rails_helper.rb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+	config.include FactoryBot::Syntax::Methods
+	config.include Warden::Test::Helpers
+end
+
+
+Shoulda::Matchers.configure do |config|
+	config.integrate do |with|
+		with.test_framework :rspec
+		with.library :rails
+	end
```
Comment (faraznaeem) [inline spec/rails_helper.rb]: Formatting is off

File: spec/spec_helper.rb [hunk 9]
```diff
@@ -0,0 +1,84 @@
+require 'capybara/rspec'
+require 'coveralls'
+Coveralls.wear_merged!
+RSpec.configure do |config|
+  # rspec-expectations config goes here. You can use an alternate
```
Comment (faraznaeem) [inline spec/spec_helper.rb]: please clean the comments

**Pull request merged**